design/one-pager-ignore-changes.md

```diff
@@ -0,0 +1,658 @@
+Providers. It will also enable the `initProvider` feature.
+
+### Examples how the new functionality would affect the use cases
+
+Some use cases mentioned in the beginning of the document would be
+solved just by omitting the management policy `LateInitialize`. For
+others, the `initProvider` field would need to be used, mostly because
+the fields in question are required on creation.
+
+#### AutoScalingGroup
+
+The `desiredCapacity` is not required on creation, but populated by late
+initialization. It can be solved just by omitting `Late Initialize` management
+policy.
+
+```yaml
+spec:
+  managementPolicy: ["Create", "Update", "Delete", "Observe"]
+  forProvider:
+    maxSize: 5
+    minSize: 1
+    launchConfigurationNameRef:
+      name: sample-launch-config
+    ...
+```
+
+#### DynamoDB Table
+
+The `readCapacity` and `writeCapacity` are not required, but they are required
+if the `billingMode` is set to `PROVISIONED`. So the `initProvider` field
+would need to be used alongside omitting `LateInitialize` management policy.
+
+```yaml
+spec:
+  managementPolicy: ["Create", "Update", "Delete", "Observe"]
+  initProvider:
+    readCapacity: 1
+    writeCapacity: 1
+  forProvider:
+    billingMode: PROVISIONED
+...
+```
+
+#### EKS NodeGroup
+
+The `scalingConfig.desiredSize` is required so `initProvider` would need
+to be used alongside omitting `LateInitialize` management policy. This way
+the autoscaler would be able to control the `desiredSize` after creation.
+
+```yaml
+spec:
+  managementPolicy: ["Create", "Update", "Delete", "Observe"]
+  initProvider:
+    scalingConfig:
+      desiredSize: 1
+  forProvider:
+    region: us-east-1
+    scalingConfig:
+      maxSize: 5
+      minSize: 1
+```
+
+#### GCP Node Pool
+
+Just omitting the `LateInitialize` management policy would be enough as the
+`version` is not required on creation.
+
+```yaml
+spec:
+  managementPolicy: ["Create", "Update", "Delete", "Observe"]
+  forProvider:
+  ...
+```
+
+#### Azure AD Group
+
+The use case where `members` and `owners` want to be set on creation and
+then ignored on updates would be solved by using `initProvider` alongside
+omitting the `LateInitialize` management policy.
+
+Example:
+```yaml
+spec:
+  managementPolicy: ["Create", "Update", "Delete", "Observe"]
+  initProvider:
+    members:
+      - user1
+      - user2
+    owners:
+      - user3
+  forProvider:
+    displayName: my-group
+    securityEnabled: true
+```
+
+#### Tags
+
+Setting initial tags should work with `initProvider`. However, it seems that
+the providers are setting some default tags in the into the `forProvider`
+field in the [Initialize] step and are updating the resource, so this would
+need to be changed to use `initProvider` instead, or we can check if those
+tags are actually needed or just skip setting them similarly to how
+ObserveOnly does it.
+
+Ref: [Upjet Initialize] or [AWS community provider tag example].
```

Comment:
I am wondering if for this use case we should:
- add a new management policy enum "Initialize" which controls if this step of the reconciler loop should be executed.
- handle this in the providers, meaning remove forcing the default tags in `spec.forProvider` . I am not aware if these tags are required for something or are they informative. In any case this behavior is currently the issue with this use case. We could remove initializing them all togather, or initialize them in `spec.initProviders`

I am closer to solution no1, adding another enum to control the initialization. But that is in part because I am not sure how solution no2 would affect the providers.

Reply:
> add a new management policy enum "Initialize" which controls if this step of the reconciler loop should be executed.

Initialize contains other initializers like [NameAsExternalName](https://github.com/crossplane/crossplane-runtime/blob/1316ae6695eec09cf47abdfd0bc6273aeaab1895/pkg/reconciler/managed/reconciler.go#L479), I don't think exposing control of this to users is a good idea. 

> We could remove initializing them all together

They were added on purpose so, not sure if we should remove them unless we see them causing a significant issue that we cannot avoid.

> initialize them in `spec.initProviders`

This could be an option but as I commented above, I am wondering if we could support `spec.initProvider` on a case-by-case basis instead of supporting it for all managed resources.

Another idea could be to make an exception for tags for now. Are there any signals that they would cause an issue? May be I am missing the problem here.

Reply:
> Another idea could be to make an exception for tags for now. Are there any signals that they would cause an issue? May be I am missing the problem here.

The issue with the tags is that as they are always present in the `spec.forProvider`, users are not able to manage the tags externally, and the proposed solution here does not help with that because they are added in the Initialize step. 

Maybe the tags problem should be handled in another issue.